### PR TITLE
WMS: use correct default bbox according to the SRS

### PR DESCRIPTION
**Pull request by mmomtchev**
## What does this PR do?

It sets the default `bbox` in the WMS driver according to the CRS (instead of a hard-coded WGS84 world extent)

## What are related issues/pull requests?

#4093 

## Tasklist

 - [x] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: gdal/frmts/wms/wmsdriver.cpp [hunk 1]
```diff
@@ -146,14 +146,44 @@ CPLXMLNode * GDALWMSDatasetGetConfigFromURL(GDALOpenInfo *poOpenInfo)
...
-        else
-            osBBOX = "-180,-90,180,90";
+        if (osBBOX.empty()) {
+            OGRSpatialReference oSRS;
+            oSRS.SetFromUserInput(osSRSValue);
+            oSRS.AutoIdentifyEPSG();
+
+            double dfWestLongitudeDeg, dfSouthLatitudeDeg,
+                    dfEastLongitudeDeg, dfNorthLatitudeDeg;
+            oSRS.GetAreaOfUse(&dfWestLongitudeDeg, &dfSouthLatitudeDeg,
```
Comment (rouault) [inline gdal/frmts/wms/wmsdriver.cpp:169]: return value should be checked

File: gdal/frmts/wms/wmsdriver.cpp [hunk 2]
```diff
@@ -146,14 +146,44 @@ CPLXMLNode * GDALWMSDatasetGetConfigFromURL(GDALOpenInfo *poOpenInfo)
...
+                    &dfEastLongitudeDeg, &dfNorthLatitudeDeg, nullptr);
+            OGRCoordinateTransformation *poCT =
```
Comment (rouault) [inline gdal/frmts/wms/wmsdriver.cpp:171]: auto poCT = std::unique_ptr<OGRCoordinateTransformation>(...)
and check return value

File: gdal/frmts/wms/wmsdriver.cpp [hunk 3]
```diff
@@ -146,14 +146,44 @@ CPLXMLNode * GDALWMSDatasetGetConfigFromURL(GDALOpenInfo *poOpenInfo)
...
+            double dfWestLongitudeDeg, dfSouthLatitudeDeg,
+                    dfEastLongitudeDeg, dfNorthLatitudeDeg;
+            oSRS.GetAreaOfUse(&dfWestLongitudeDeg, &dfSouthLatitudeDeg,
+                    &dfEastLongitudeDeg, &dfNorthLatitudeDeg, nullptr);
+            OGRCoordinateTransformation *poCT = 
+                    OGRCreateCoordinateTransformation(OGRSpatialReference::GetWGS84SRS(),
+                                                        &oSRS);
+            poCT->Transform(1, &dfWestLongitudeDeg, &dfNorthLatitudeDeg);
+            poCT->Transform(1, &dfEastLongitudeDeg, &dfSouthLatitudeDeg);
+            const double dfMaxX = std::max(dfWestLongitudeDeg, dfEastLongitudeDeg);
```
Comment (rouault) [inline gdal/frmts/wms/wmsdriver.cpp:176]: ``#include <algorithm>`` to add toward top of the file

File: gdal/frmts/wms/wmsdriver.cpp (hunk 1 above)
Comment (rouault) [reply gdal/frmts/wms/wmsdriver.cpp:170]: there's just the testing of the return value of GetAreaOfUse() that's missing

Comment (rouault) [reply gdal/frmts/wms/wmsdriver.cpp:170]: @mmomtchev ping w.r.t my above remark

File: autotest/gdrivers/wms.py [hunk 4]
```diff
@@ -634,6 +634,11 @@ def test_wms_16():
     ds = gdal.Open(name)
     assert ds is not None, ('open of %s failed.' % name)
 
+    # check that the default bbox works for srs != EPSG:4326
+    name = 'http://demo.opengeo.org/geoserver/wms?SERVICE=WMS&request=GetMap&version=1.1.1&layers=og:bugsites&styles=&srs=EPSG:26713'
```
Comment (jratike80) [inline autotest/gdrivers/wms.py:638]: That server does not exist, how this test can pass?

**Pull request merged**